src/tests/test_pyodide_http.py

```diff
@@ -80,27 +80,38 @@ def raise_for_status_fixture(httpserver):
 
 
 @run_in_pyodide
-async def test_pyfetch_raise_for_status(selenium, raise_for_status_fixture):
+async def test_pyfetch_raise_for_status_does_not_raise_200(
+    selenium, raise_for_status_fixture
+):
     import pytest
 
-    from pyodide.http import pyfetch
+    from pyodide.http import HttpStatusError, pyfetch
 
     resp = await pyfetch(raise_for_status_fixture["/status_200"])
     resp.raise_for_status()
     assert await resp.string() == "Some data here!"
 
     resp = await pyfetch(raise_for_status_fixture["/status_404"])
     with pytest.raises(
-        OSError, match="404 Client Error: NOT FOUND for url: .*/status_404"
-    ):
+        HttpStatusError, match="404 Client Error: NOT FOUND for url: .*/status_404"
+    ) as error_404:
         resp.raise_for_status()
 
+    assert error_404.status == 404
```

Comment:
Need to put `.value` here and in all of these places:
```suggestion
    assert error_404.value.status == 404
```